### PR TITLE
sim/posix: Add the host_system interface used to execute the host command

### PR DESCRIPTION
**Pull request by liqinhuixm**
## Summary
Encapsulate the host system/popen interface to host_system.
## Impact
We can use the host_system interface to execute the host command.
## Testing
  - Execute the `ls`,  do not receive the return result of the `ls` command.
    ret = host_system(NULL, 0, "%s", "ls");
  - Execute the `ls`,  receive the return result of the `ls` command with a buffer of length `buf_size`.
    ret = host_system(buf, buf_size, "%s", "ls");

### CODE REVIEW

**Review by xiaoxiang781216 — APPROVED** [review]

**Review by yamt — COMMENTED** [review]
what's the expected user of this?

**Comment by xiaoxiang781216** [comment]
> what's the expected user of this?

sim wifi simulator need call wpa supplicant through host_system.

**Comment by yamt** [comment]
> > what's the expected user of this?
> 
> sim wifi simulation need call wpa supplicant through host_system.

is it for device drivers?

File: arch/sim/src/sim/posix/sim_hostmisc.c [hunk 1]
```diff
@@ -108,6 +109,61 @@ int host_backtrace(void** array, int size)
 #endif
 }
 
+/****************************************************************************
+ * Name: host_system
```
[the remark this answers is not in the document]
Comment (xiaoxiang781216) [reply arch/sim/src/sim/posix/sim_hostmisc.c:113]: do you want the caller call host_read/host_close by self? It's a common case to call system and gather the output. Exposing host_popen will make the caller duplicate the similar code in many places.

**Comment by xiaoxiang781216** [comment]
> > > what's the expected user of this?
> > 
> > 
> > sim wifi simulation need call wpa supplicant through host_system.
> 
> is it for device drivers?

yes, net wifi driver will call host_system("wpaxxx") to interact with host's WiFi.

File: arch/sim/src/sim/posix/sim_hostmisc.c (hunk 1 above)
[the remark this answers is not in the document]
Comment (yamt) [reply arch/sim/src/sim/posix/sim_hostmisc.c:113]: ok. i prefer to name it something not `system` if it isn't system(). but i have no strong opinions.

**Comment by yamt** [comment]
> > > > what's the expected user of this?
> > > 
> > > 
> > > sim wifi simulation need call wpa supplicant through host_system.
> > 
> > 
> > is it for device drivers?
> 
> yes, net wifi driver will call host_system("wpaxxx") to interact with host's WiFi.

ok. it makes sense.

**Review by yamt — APPROVED** [review]

File: arch/sim/src/sim/posix/sim_hostmisc.c (hunk 1 above)
[the remark this answers is not in the document]
Comment (xiaoxiang781216) [reply arch/sim/src/sim/posix/sim_hostmisc.c:113]: it's unlucky that the POSIX spec doesn't support system return the output from application, which is a very common requirment.

**Pull request merged**